integration-tests/features/steps/component_analysis_v2.py

```diff
@@ -0,0 +1,163 @@
+"""Tests for API endpoints that performs v2 component analysis."""
+import requests
+
+import time
+
+from behave import then, when
+from urllib.parse import urljoin
+
+from src.parsing import parse_token_clause
+
+
+def component_analysis_v2_url(context, ecosystem, component, version):
+    """Construct URL for the component analyses REST API call."""
+    return urljoin(context.core_v2_api_url,
+                   '/api/v2/component-analyses/{e}/{c}/{v}'.format(e=ecosystem,
+                                                                   c=component,
+                                                                   v=version))
+
+
+def perform_v2_component_analysis(context, ecosystem, package, version, use_user_key, rate):
+    """Call API endpoint to v2 analysis for component."""
+    context.duration = None
+    start_time = time.time()
+    url = component_analysis_v2_url(context, ecosystem, package, version)
+    if use_user_key:
+        for _ in range(rate):
+            context.response = requests.get(
+                url, params={'user_key': context.three_scale_preview_user_key})
+            # break the loop if rate limit exceeded
+            if context.response.status_code == 429:
+                break
+    else:
+        context.response = requests.get(url)
+    end_time = time.time()
+    # compute the duration
+    # plase note that duration==None in case of any errors (which is to be expected)
+    context.duration = end_time - start_time
+
+
+@when("I start v2 component analyses {ecosystem}/{package}/{version} {user_key} user_key")
+@when("I start v2 component analyses {ecosystem}/{package}/{version} {rate:d} "
+      "times in a minute {user_key} user_key")
+def start_v2_component_analysis(context, ecosystem, package, version, user_key, rate=1):
+    """Analyse the given component."""
+    use_user_key = parse_token_clause(user_key)
+    perform_v2_component_analysis(context, ecosystem, package, version, use_user_key, rate)
+
+
+@then('I should find no recommendation')
+def check_recommendation_in_result(context):
+    """Check if no recommendation comes."""
```

Comment:
I dont think this how the current API is working. If we dont have a recommendation it is sending an empty string.

Reply:
it just sends a response like `recommendation : []`